src/asl.ts

```diff
@@ -4871,13 +5016,331 @@ export namespace ASLGraph {
           }),
     };
   };
+
+  /**
+   * Compare any two {@link ASLGraph.Output} values.
+   */
+  export const compareOutputs = (
+    leftOutput: ASLGraph.Output,
+    rightOutput: ASLGraph.Output,
+    operator: ASL.ValueComparisonOperators | "!=" | "!=="
+  ): Condition => {
+    if (
+      ASLGraph.isLiteralValue(leftOutput) &&
+      ASLGraph.isLiteralValue(rightOutput)
+    ) {
+      return ((operator === "==" || operator === "===") &&
+        leftOutput.value === rightOutput.value) ||
+        ((operator === "!=" || operator === "!==") &&
+          leftOutput.value !== rightOutput.value) ||
+        (leftOutput.value !== null &&
+          rightOutput.value !== null &&
+          ((operator === ">" && leftOutput.value > rightOutput.value) ||
+            (operator === "<" && leftOutput.value < rightOutput.value) ||
+            (operator === "<=" && leftOutput.value <= rightOutput.value) ||
+            (operator === ">=" && leftOutput.value >= rightOutput.value)))
+        ? ASL.trueCondition()
+        : ASL.falseCondition();
+    }
+
+    const [left, right] =
+      ASLGraph.isJsonPath(leftOutput) || ASLGraph.isConditionOutput(leftOutput)
+        ? [leftOutput, rightOutput]
+        : [
+            rightOutput as ASLGraph.JsonPath | ASLGraph.ConditionOutput,
+            leftOutput,
+          ];
+    // if the right is a variable and the left isn't, invert the operator
+    // 1 >= a -> a <= 1
+    // a >= b -> a >= b
+    // a >= 1 -> a >= 1
+    const op = leftOutput === left ? operator : invertBinaryOperator(operator);
+
+    return ASLGraph.compare(left, right, op as any);
+  };
+
+  export const compare = (
+    left: ASLGraph.JsonPath | ASLGraph.ConditionOutput,
+    right: ASLGraph.Output,
+    operator: ASL.ValueComparisonOperators | "!=" | "!=="
+  ): Condition => {
```

Comment:
no `const arrow function` please. It reads backwards.

Reply:
These are in the namespace and exported, so it really doesn't matter, but I did change all of the const arrow functions in the namesapces to be functions.